src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -148,9 +148,430 @@
+                break;
+
+            case CHAR_CURRENCY:
+                if (!currencyAdded)
+                {
+                    currencyAdded = true;
+                    destPattern[index++] = 'C';
+                }
+                break;
+
+            case CHAR_SPACE:
+            case CHAR_NBSPACE:
+                if (!spaceAdded)
+                {
+                    spaceAdded = true;
+                    destPattern[index++] = ' ';
+                }
+                break;
+
+            case CHAR_MINUS:
+            case CHAR_OPENPAREN:
+            case CHAR_CLOSEPAREN:
+                minusAdded = true;
+                destPattern[index++] = (char)ch;
+                break;
+
+            case CHAR_PERCENT:
+                destPattern[index++] = '%';
+                break;
+        }
+    }
+
+    return destPattern;
+}
+
+/*
+Function:
+GetNumericPattern
+
+Determines the pattern from the decimalFormat and returns the matching pattern's
+index from patterns[].
+Returns index -1 if no pattern is found.
+*/
+static int GetNumericPatternNative(const char* pNumberFormat,
+                                   const char* patterns[],
+                                   int patternsCount,
+                                   int isNegative)
+{
+    const int INVALID_FORMAT = -1;
+    const int MAX_DOTNET_NUMERIC_PATTERN_LENGTH = 6; // example: "(C n)" plus terminator
+    char* normalizedPattern = NormalizeNumericPattern(pNumberFormat, isNegative);
+
+    if (!normalizedPattern)
+    {
+        return INVALID_FORMAT;
+    }
+
+    size_t normalizedPatternLength = strlen(normalizedPattern);
+
+    assert(normalizedPatternLength > 0);
+    assert(normalizedPatternLength < MAX_DOTNET_NUMERIC_PATTERN_LENGTH);
+
+    if (normalizedPatternLength == 0 || normalizedPatternLength >= MAX_DOTNET_NUMERIC_PATTERN_LENGTH)
+    {
+        free(normalizedPattern);
+        return INVALID_FORMAT;
+    }
+
+    for (int i = 0; i < patternsCount; i++)
+    {
+        if (strcmp(normalizedPattern, patterns[i]) == 0)
+        {
+            free(normalizedPattern);
+            return i;
+        }
+    }
+
+    assert(false); // should have found a valid pattern
+
+    free(normalizedPattern);
+    return INVALID_FORMAT;
+}
+
+static int32_t GetValueForNumberFormat(NSLocale *currentLocale, LocaleNumberData localeNumberData)
+{
+    NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];            
+    numberFormatter.locale = currentLocale;
+    const char *pFormat;
+    int32_t value;
+
+    switch(localeNumberData)
+    {
+        case LocaleNumber_PositiveMonetaryNumberFormat:
+        {
+            numberFormatter.numberStyle = NSNumberFormatterCurrencyStyle;
+            static const char* Patterns[] = {"Cn", "nC", "C n", "n C"};
+            pFormat = [[numberFormatter positiveFormat] UTF8String];
+            value = GetNumericPatternNative(pFormat, Patterns, sizeof(Patterns)/sizeof(Patterns[0]), false);
```

Comment:
yes they are tied together.

Reply:
Regarding patternsCount, when it is pushed to the function I get `warning: sizeof on array function parameter will return size of 'const char **' instead of 'const char *[]' [-Wsizeof-array-argument]`. 
Arrays are always passed to functions as pointers, so I'm not sure how can I calculate patternsCount in that case.

Reply:
Ah right good point. I think its fine to pass in the size of the array